Code/GraphMol/DistGeomHelpers/Embedder.cpp

```diff
@@ -749,11 +809,16 @@ bool embedPoints(RDGeom::PointPtrVect *positions, detail::EmbedArgs eargs,
         gotCoords = EmbeddingOps::minimizeWithExpTorsions(*positions, eargs,
                                                           embedParams);
       }
-      // test if chirality is correct
-      if (embedParams.enforceChirality && gotCoords &&
-          (eargs.chiralCenters->size() > 0)) {
-        gotCoords =
-            EmbeddingOps::finalChiralChecks(positions, eargs, embedParams);
+      // test if stereo is correct
+      if (embedParams.enforceChirality && gotCoords) {
+        if (!eargs.chiralCenters->empty()) {
+          gotCoords =
+              EmbeddingOps::finalChiralChecks(positions, eargs, embedParams);
+        }
+        if (gotCoords && !eargs.stereoDoubleBonds->empty()) {
```

Comment:
I thought about this and couldn't decide, so I took the easiest route. :shrug: 

We could change this to two tests: one which always fails if there's a linear arrangement at the end a double bond and another which checks that cis/trans are "correct" and which is only enabled if checkChirality is true.
I'm reasonably sure that there's no case where you'd want the linear double bonds, but...

Reply:
To me, it would make sense to distinguish between catching an incorrect linear arrangement, and the cis/trans check.

Reply:
> To me, it would make sense to distinguish between catching an incorrect linear arrangement, and the cis/trans check.

@sriniker so do you mean that the test for the linear arrangement should always be done?

Reply:
Yes, I would do that because it's basically like a bug